x-pack/filebeat/input/s3/s3_integration_test.go

```diff
@@ -0,0 +1,417 @@
+	}
+
+	if accessKeyID == "" {
+		info = "Skipping: $AWS_ACCESS_KEY_ID or $AWS_PROFILE_NAME not set or set to empty"
+	} else if secretAccessKey == "" {
+		info = "Skipping: $AWS_SECRET_ACCESS_KEY not set or set to empty"
+	} else {
+		awsConfig = awscommon.ConfigAWS{
+			AccessKeyID:     accessKeyID,
+			SecretAccessKey: secretAccessKey,
+		}
+
+		if sessionToken != "" {
+			awsConfig.SessionToken = sessionToken
+		}
+	}
+	config := config{
+		QueueURL:          queueURL,
+		AwsConfig:         awsConfig,
+		VisibilityTimeout: visibilityTimeout,
+	}
+	return config, info
+}
+
+func testSetup(t *testing.T) context.CancelFunc {
+	t.Helper()
+	_, cancel := context.WithCancel(context.Background())
+	return cancel
+}
+
+func uploadSampleLogFile(t *testing.T, bucketName string, svcS3 s3iface.ClientAPI) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	file, err := os.Open(fileDir)
+	if err != nil {
+		t.Fatalf("Failed to open file %v", fileDir)
+		os.Exit(1)
+	}
+	defer file.Close()
+
+	s3PutObjectInput := s3.PutObjectInput{
+		Bucket: aws.String(bucketName),
+		Key:    aws.String(filepath.Base(fileDir)),
+		Body:   file,
+	}
+	req := svcS3.PutObjectRequest(&s3PutObjectInput)
+	output, err := req.Send(ctx)
+	if err != nil {
+		t.Fatalf("failed to put object into s3 bucket: %v", output)
+	}
+}
+
+func collectOldMessages(t *testing.T, queueURL string, visibilityTimeout int64, svcSQS sqsiface.ClientAPI) []string {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// receive messages from sqs
+	req := svcSQS.ReceiveMessageRequest(
+		&sqs.ReceiveMessageInput{
+			QueueUrl:              &queueURL,
+			MessageAttributeNames: []string{"All"},
+			MaxNumberOfMessages:   &maxNumberOfMessage,
+			VisibilityTimeout:     &visibilityTimeout,
+			WaitTimeSeconds:       &waitTimeSecond,
+		})
+
+	output, err := req.Send(ctx)
+	if err != nil {
+		t.Fatalf("failed to receive message from SQS: %v", output)
+	}
+
+	var oldMessageHandles []string
+	for _, message := range output.Messages {
+		oldMessageHandles = append(oldMessageHandles, *message.ReceiptHandle)
+	}
+
+	return oldMessageHandles
+}
+
+func (input *s3Input) deleteAllMessages(t *testing.T, awsConfig awssdk.Config, queueURL string, visibilityTimeout int64, svcSQS sqsiface.ClientAPI) error {
+	var messageReceiptHandles []string
+	init := true
+	for init || len(messageReceiptHandles) > 0 {
+		init = false
+		messageReceiptHandles = collectOldMessages(t, queueURL, visibilityTimeout, svcSQS)
+		for _, receiptHandle := range messageReceiptHandles {
+			err := input.deleteMessage(queueURL, receiptHandle, svcSQS)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func defaultTestConfig() *common.Config {
+	return common.MustNewConfigFrom(map[string]interface{}{
+		"queue_url": os.Getenv("QUEUE_URL"),
+	})
+}
+
+func runTest(t *testing.T, cfg *common.Config, run func(input *s3Input, out *stubOutleter, t *testing.T)) {
```

Comment:
Usually `testing.T` is the first argument on tests
```suggestion
func runTest(t *testing.T, cfg *common.Config, run func(t *testing.T, input *s3Input, out *stubOutleter)) {
```